mapper/cumulocity/c8y_smartrest/src/smartrest_deserializer.rs

```diff
@@ -164,6 +164,57 @@ impl SmartRestUpdateSoftwareModule {
     }
 }
 
+fn to_datetime<'de, D>(deserializer: D) -> Result<DateTime<FixedOffset>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    // NOTE `NaiveDateTime` is used here because c8y uses for log requests a date time string which
+    // does not exactly equal `chrono::DateTime::parse_from_rfc3339`
+    // c8y result:
+    // 2021-10-23T19:03:26+0100
+    // rfc3339 expected:
+    // 2021-10-23T19:03:26+01:00
+    // so we add a ':'
+    let mut date_string: String = Deserialize::deserialize(deserializer)?;
+    let str_size = date_string.len();
+    date_string = date_string[0..str_size - 2].to_string()
+        + ":"
+        + &date_string[str_size - 2..str_size].to_string();
+    Ok(DateTime::parse_from_rfc3339(&date_string).unwrap())
```

Comment:
An unwrap to be removed, I suppose it may not be straightforward, but bare in mind that if parse will fail the app will panic.